### DISCUSSION TITLE
Mp3 player functions - Autoplay - Repeat - Shuffle to apply to the folders

### DISCUSSION THREAD

**Discussion by Officepoint** [rejetto/hfs#478] [General]
Hi!

Can these three mp3 player functions - Autoplay - Repeat - Shuffle to apply to the folders?
For example, the Audio folder contains two other folders - Songs 1 and Songs 2, and after the songs in Songs 1 folder are finished, it automatically switches to folder Songs 2, when the Autoplay button is active.
If I activate the Shuffle button, then the mp3 player chooses the songs from the two folders Songs 1 and Songs 2 by itself, the same for the Repeat button.

Thank you!

**Comment by rejetto** [comment]
you'll find these features in 0.52, already published as alpha

**Comment by rejetto** [comment]
and these features work also on search result, so you can shuffle mp3s of all folders, if you first serach .mp3

**Comment by Officepoint** [comment]
Hi Rejjeto!

I tried in HFS 0.52.alpha2 but the Repeat - Shuffle buttons do not appear, they are not visible, and mp3 player and not go to the next song in the next folder, with button Autoplay activated.
I tried in HFS 0.52.alpha3 and the buttons appear but after a few tests they disappeared.
The video is for HFS 0.52.alpha2.
I did test to see if the mp3 player will go to the next song in the next folder, namely Songs 2, but it doesn't go any further.
Possibly I'm wrong somewhere but I can't figure it out. 

https://github.com/rejetto/hfs/assets/94564442/5dd1bd63-fb61-4f49-8a7d-6842a902fa60

Thanks!

**Comment by rejetto** [comment]
i've published alpha3 some days ago, but something is funny and it's NOT showing in the list of releases :(
and yet it exists
<img width="424" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/900ce3cf-b8d4-4017-a43f-8e47f5d29c65">

i'll try to see if i can work around this problem
in the meantime you can access the page directly at
https://github.com/rejetto/hfs/releases/tag/v0.52.0-alpha3
and make an automatic update right clicking on the check-for-updates button and pasting your link, like if you are on windows it is https://github.com/rejetto/hfs/releases/download/v0.52.0-alpha3/hfs-windows.zip

**Comment by rejetto** [comment]
it was hard but i've managed to fix it (hopefully)
<img width="207" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/dae60087-f688-49b6-8d20-4018f2b4a391">

**Comment by rejetto** [comment]
so alpha3 has been "not" out for one week, and now i've just published beta4, that gives you flac and id3 tags support

**Comment by Officepoint** [comment]
Hi!
I tested version 0.52.0-beta4, and now the Repeat - Shuffle buttons are functional, but the mp3 player does not go to the next song from the next folder when the Autoplay button is activated.
Thanks!

**Comment by rejetto** [comment]
i don't see why it should 😀 it's not a media player.
but you can get a similar result, go in the parent folder, and search *, or .mp3
the file-show will work on the list of results, so it will continue cross different folders.

**Reply by ZIKVLD** [reply]
Everything is fine, but the covers stop working

**Comment by Officepoint** [comment]
> i don't see why it should 😀 it's not a media player. but you can get a similar result, go in the parent folder, and search *, or .mp3 the file-show will work on the list of results, so it will continue cross different folders.

Hi Rejjeto!
I wanted to create a folder with all the songs as a Playlist for those who wanted to listen all the songs without browsing through folders. I can copy all the songs again into one folder but that way I would lose space on my hdd because I would have the songs twice. If mp3 player knew how to move from folder to folder no user intervention would be needed.
The solution with the Search function is good, except that I will have to specify to each user how to do that, i mean, how to use Search function, each time he wants to listen all songs from all folders. 
Anyway, thanks for information!
All the best!

**Comment by rejetto** [comment]
i wouldbe easy if you could provide a direct link to the parent folder with the search already done.
I will work to make this possible.

**Comment by rejetto** [comment]
[beta4.2](https://github.com/rejetto/hfs/files/14259038/hfs-windows.zip)

with this version, you can make a link to the folder and add manually `?search=.mp3|.jpg`

the .jpg is if you want to have covers too

**Reply by ZIKVLD** [reply]
Yes it works

**Reply by Officepoint** [reply]
Hi!
For the moment I can't figure out how to add the link and extensions ?search=.mp3
You mean I can add the parent folder that contains all the other mp3 folders, then create a web-link from Add>web-link? 

Thanks!

**Reply by rejetto** [reply]
you can do that if what you want the link to be in the hfs page, but you can also give the link sending an email

**Reply by Officepoint** [reply]
What i'm doing wrong?

https://github.com/rejetto/hfs/assets/94564442/e49dd126-a78c-44a4-861e-9d670b2a2901

**Reply by rejetto** [reply]
TBH, i don't know. I made the same things and it's working.
Well, i have a different problem, it starts listing all my mp3s and never stops, but I'm quite sure it has nothing to do with your problem.

**Comment by rejetto** [comment]
[beta4.21](https://github.com/rejetto/hfs/files/14272223/hfs-windows.zip)

i made a fix, but i don't know if it will solve your problem too

**Reply by ZIKVLD** [reply]
Yes it works

**Comment by rejetto** [comment]
the link you create doesn't need to include the IP part

**Comment by Officepoint** [comment]
Hi!
If there is a file called cover.jpg, or albumart.jpg in the song folder, when the player reaches the last song it sometimes displays an X with the message cover failed. Most of the time when it reaches the last song it repeats the same song over and over again without taking it from the beginning of the song list. Other times after the last song has been played, it restarts and after a 2 second interval starts the next song, playing them both at the same time.
In the first video you can see at the end how it overlaps the pictures and plays two songs at the same time. Unfortunately on this video I forgot to turn the sound on when i recorded

https://github.com/rejetto/hfs/assets/94564442/bd0beff5-e6fc-4595-8c4c-2aff7222b726

https://github.com/rejetto/hfs/assets/94564442/9ede4a2f-59ac-4ed6-9930-b4bebab1e1df

Thanks!

**Comment by rejetto** [comment]
thanks for your videos.
Please, see if this is fixing: [beta4.3](https://github.com/rejetto/hfs/files/14310406/hfs-windows.zip)

**Comment by Officepoint** [comment]
Unfortunately, this version is not working at all.

https://github.com/rejetto/hfs/assets/94564442/cce1501d-122b-4525-b4dc-c4d84d142778

**Reply by rejetto** [reply]
sorry, i couldn't see the problem on my side, but now i do

**Comment by Officepoint** [comment]
No problem Rejetto, if i can, i am glad to help.

All the best!

**Comment by rejetto** [comment]
ok, this should fix that
[beta4.31](https://github.com/rejetto/hfs/files/14318270/hfs-windows.zip)

**Comment by Officepoint** [comment]
Hi Rejetto!

This version skips songs, doesn't load the image sometimes and gets stuck on a song when I press the arrow to play the next song.
Thanks!


https://github.com/rejetto/hfs/assets/94564442/f384f1f6-860c-40f2-bc83-6df5664e01c5

**Reply by rejetto** [reply]
thanks for your video

**Comment by rejetto** [comment]
the "image" problem, i don't see it in your video: your cover is in "song 3", so it will only be used for song3's files.

i'm not sure "stuck" problem, but i had to quite change the algorithm because of the "skipping first" problem, so please give this a try
[beta4.32](https://github.com/rejetto/hfs/files/14321864/hfs-windows.zip)

**Comment by Officepoint** [comment]
Yes, the cover it is only in folder Songs 3.
It seems that now the problem only manifests itself in the folder where the image exists ( Songs 3 ), otherwise it works ok, at least as far as I tested.
I don't know how it would behave with a few hundred songs or even more.

https://github.com/rejetto/hfs/assets/94564442/1448c105-25a7-4f93-8f38-ac84bb3f767a

Thanks!

**Reply by rejetto** [reply]
thanks for your feedback

**Comment by rejetto** [comment]
i couldn't reproduce the problem you are having with that folder.
maybe you can send me a zip with those files and i can see if it happens to me too?
also, press F12 and see if you find anything strange in console.

**Comment by Officepoint** [comment]
It seems that if I press F12 in the browser, with the window resized the mp3 player no longer skips songs, although it still displays the image at the end, but then starts the songs from the beginning.

https://github.com/rejetto/hfs/assets/94564442/42eacde0-4e0d-416e-b8ba-8d5daaf7ebe9

[Songs 3.zip](https://github.com/rejetto/hfs/files/14326175/Songs.3.zip)

**Comment by rejetto** [comment]
hey, sorry i forgot about you :D
i tried with the zip you sent, and it's working perfectly on my computer, look
https://github.com/rejetto/hfs/assets/1367199/1e1b5f77-3bc5-48b6-a680-dac15c41dcb3

so it's not the content of the folder.
also, you seem to use Chrome, and without extensions, so that's not it.
I also tried in incognito, to exclude my chrome extensions.
i still have no clue on what's causing your problem.
